packages/js/src/feeds/feeds.ts

```diff
@@ -1,33 +1,187 @@
+  async fetch({ page = 0, status, ...restOptions }: FetchFeedArgs = {}): Promise<PaginatedResponse<Notification>> {
     return this.callWithSession(async () => {
+      const args = { page, status, ...restOptions };
       try {
-        this._emitter.emit('feeds.fetch.pending');
+        this._emitter.emit('feeds.fetch.pending', { args });
 
-        const response = await this._apiService.getNotificationsList(page, restOptions);
+        const response = await this._apiService.getNotificationsList(page, {
+          ...restOptions,
+          ...(status && SEEN_OR_UNSEEN.includes(status) && { seen: status === NotificationStatus.SEEN }),
+          ...(status && READ_OR_UNREAD.includes(status) && { seen: status === NotificationStatus.READ }),
+        });
         const modifiedResponse: PaginatedResponse<Notification> = {
           ...response,
-          data: response.data.map((el) => new Notification(el)),
+          data: response.data.map((el) => new Notification(el as TODO)),
         };
 
-        this._emitter.emit('feeds.fetch.success', { response: modifiedResponse });
+        this._emitter.emit('feeds.fetch.success', { args, result: modifiedResponse });
+
+        return modifiedResponse;
+      } catch (error) {
+        this._emitter.emit('feeds.fetch.error', { args, error });
+        throw error;
+      }
+    });
+  }
+
+  async fetchCount({ feedIdentifier, status = NotificationStatus.UNSEEN }: FetchCountArgs = {}): Promise<number> {
+    return this.callWithSession(async () => {
+      const args = { feedIdentifier, status };
+      try {
+        this._emitter.emit('feeds.fetch_count.pending', { args });
+
+        let response: { count: number };
+        if (SEEN_OR_UNSEEN.includes(status)) {
+          response = await this._apiService.getUnseenCount({
+            feedIdentifier,
+            seen: status === NotificationStatus.SEEN,
+          });
+        } else {
+          response = await this._apiService.getUnreadCount({
+            feedIdentifier,
+            read: status === NotificationStatus.READ,
+          });
+        }
+
+        this._emitter.emit('feeds.fetch_count.success', { args, result: response.count });
+
+        return response.count;
+      } catch (error) {
+        this._emitter.emit('feeds.fetch_count.error', { args, error });
+        throw error;
+      }
+    });
+  }
+
+  async markNotificationAs(args: MarkNotificationAsByIdArgs): Promise<Notification>;
+  async markNotificationAs(args: MarkNotificationAsByInstanceArgs): Promise<Notification>;
+  async markNotificationAs(args: MarkNotificationAsArgs): Promise<Notification> {
+    return this.callWithSession(async () =>
+      markNotificationAs({
+        emitter: this._emitter,
+        apiService: this._apiService,
+        args,
+      })
+    );
+  }
+
+  async markNotificationsAs(args: MarkNotificationsAsByIdsArgs): Promise<Notification[]>;
```

Comment:
I suggest we go with a specific method on this public API for better DX, such as `novu.feeds.markAllAsRead()`.

Reply:
If you are saying to go additionally with these then yes we can introduce this granularity, but need to be consistent with that pattern and implement also for other methods:
- for single notification: `markNotificationAsRead, markNotificationAsUnread, markNotificationAsSeen, markNotificationAsUnseen`
- for selected notifications: `markNotificationsAsRead, markNotificationsAsUnread, markNotificationsAsSeen, markNotificationsAsUnseen`
- for all notifications: `markAllAsRead, markAllAsUnread, markAllAsSeen, markAllAsUnseen`
- similar pattern to actions